src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java

```diff
@@ -203,8 +210,47 @@ private static boolean shouldSchedulePatronNotices(LostItemFeeRefundContext cont
     final Result<CqlQuery> fetchQuery = exactMatch("loanId", context.getLoan().getId())
       .combine(exactMatchAny("feeFineType", feeFineTypes), CqlQuery::and);
 
-    return accountRepository.findAccountsAndActionsForLoanByQuery(fetchQuery)
-      .thenApply(r -> r.map(context::withAccounts));
+      return accountRepository.findAccountsAndActionsForLoanByQuery(fetchQuery)
+        .thenApply(r -> r.map(this::filterAccountsForRefund))
+        .thenApply(r -> r.map(context::withAccounts));
+  }
+
+  private Collection<Account> filterAccountsForRefund(Collection<Account> accounts) {
+    Account latestLostItemFeeAccount = getLatestAccount(accounts, LOST_ITEM_FEE_TYPE);
+
+    if (latestLostItemFeeAccount != null
+      && latestLostItemFeeAccount.getPaymentStatus() != null
+      && latestLostItemFeeAccount.getCreationDate() != null
+      && !latestLostItemFeeAccount.getPaymentStatus().contains(CANCELLED_PAYMENT_STATUS)) {
+
+      Collection<Account> filteredAccounts = new ArrayList<>();
```

Comment:
Why a `Collection` instead of `List`?

Reply:
It keeps the type which is passed from the previous to the next methods